### DISCUSSION TITLE
CSI Parameter overwrite?

### DISCUSSION THREAD

**Discussion by rcampos2029** [rook/rook#8952] [Q&A] [answered]
https://github.com/rook/rook/blob/67fa72aaeba86e2e360377691493d8f6caa8a445/pkg/operator/ceph/csi/controller.go#L140

The CSI controller loop watches ConfigMaps and CephClusters. When a CephCluster triggers the reconcile loop, this Get call will fail since the name in the request object is that of the CephCluster, not the ConfigMap, leading to default values being used. Doesn't this result in the default parameters being used regardless of what is in the ConfigMap?

**Comment by Madhu-1** [comment, marked as answer]
cc @leseb

**Reply by leseb** [reply]
I think this is only true if the cephcluster is in a different namespace than the Operator. I thought I was forcing the NamespacedName somewhere to be the Operator one for this CM. Looks like I was not. Also, because the predicate first searches for the CM name it will get precedence over the cephCluster condition. See: https://github.com/rook/rook/blob/67fa72aaeba86e2e360377691493d8f6caa8a445/pkg/operator/ceph/csi/predicate.go#L36-L40
It really depends on the scenario, we have:

* operator is running and there is no CM yet and no cluster
* operator is running with a CM and no cluster yet

But I agree we should force the operator's NamespacedName in case the cluster is in a different namespace than the operator.
Is this your scenario?

**Reply by rcampos2029** [reply]
Yes, I came across this while trying to figure out how to support multiple namespaces for the multus pr #8686, because I want the loop to watch and reconcile the multus holder daemonsets too. This ConfigMap should be in the operator namespace, correct? Would there be an issue in changing the Get call to explicitly get the configmap with the assumed name and namespace? Here's what I mean:
```
err = r.client.Get(r.opManagerContext, types.NamespacedName{
                Namespace: r.opConfig.OperatorNamespace,
                Name:      opcontroller.OperatorSettingConfigMapName,
        }, opConfig)
```

**Reply by leseb** [reply]
Not a problem, I did that as part of the same PR here https://github.com/rook/rook/blob/ee0759b485f84263476949bba38dfb02ed8d9658/pkg/operator/ceph/object/bucket/controller.go#L131-L134 and missed adding this in the csi code. LGTM.

**Reply by leseb** [reply]
I've opened https://github.com/rook/rook/issues/8958 if you want to send a PR for this. Thanks!

**Reply by rcampos2029** [reply]
Cool, will do
